lib/glific/saas/queries.ex

```diff
@@ -304,4 +399,129 @@ defmodule Glific.Saas.Queries do
+    cond do
+      empty(value) ->
+        dgettext("error", "Billing frequency cannot be empty.")
+        |> error(result, :billing_frequency)
+
+      value not in ["yearly", "monthly", "quarterly"] ->
+        dgettext("error", "Value should be one of yearly, monthly, or quarterly.")
+        |> error(result, :billing_frequency)
+
+      true ->
+        result
+    end
+  end
+
+  @spec validate_finance_poc(map(), map()) :: map()
+  defp validate_finance_poc(%{is_valid: false} = result, _params), do: result
+
+  defp validate_finance_poc(result, params) do
+    result
+    |> validate_text_field(params["name"], :finance_poc_name, {1, 25})
+    |> validate_text_field(params["designation"], :finance_poc_designation, {1, 25})
+    |> validate_phone(params["phone"], :finance_poc_phone)
+    |> validate_email(params["email"], :finance_poc_email)
+  end
+
+  @spec validate_submitter_details(map(), map()) :: map()
+
+  defp validate_submitter_details(%{is_valid: false} = result, _params), do: result
+
+  defp validate_submitter_details(result, params) do
+    result
+    |> validate_text_field(params["name"], :submitter_name, {1, 25})
+    |> validate_email(params["email"], :submitter_name)
+  end
+
+  @spec validate_signer_details(map(), map()) :: map()
+
+  defp validate_signer_details(%{is_valid: false} = result, _params), do: result
+
+  defp validate_signer_details(result, params) do
+    result
+    |> validate_text_field(params["name"], :signer_name, {1, 25})
+    |> validate_text_field(params["designation"], :signer_designation, {1, 25})
+    |> validate_email(params["email"], :signer_email)
+  end
+
+  @spec validate_org_details(map(), map()) :: map()
+  defp validate_org_details(%{is_valid: false} = result, _params), do: result
+
+  defp validate_org_details(result, params) do
+    result
+    |> validate_text_field(params["gstin"], :gstin, {15, 15}, true)
+    |> validate_text_field(
+      params["registered_address"],
+      :registered_address,
+      {1, 300}
+    )
+    |> validate_text_field(params["current_address"], :current_address, {0, 300})
+  end
+
+  defp parse_params(result, value, key) when is_binary(value) do
+    case Jason.decode(value) do
+      {:ok, value} ->
+        {result, value}
+
+      {:error, _} ->
+        dgettext("error", "Failed to parse the input.")
+        |> error(result, key)
+        |> then(&{&1, value})
+    end
+  end
+
+  defp parse_params(result, value, _key), do: {result, value}
+
+  @spec validate_booleans(map(), String.t(), any()) :: map()
+  defp validate_booleans(result, key, value) do
+    case value do
+      value when is_boolean(value) ->
+        Map.put(result, key, value)
+
+      "true" ->
+        Map.put(result, key, true)
+
+      "false" ->
+        Map.put(result, key, false)
+
+      _ ->
+        dgettext("error", "%{key} should be of type boolean.", key: key)
+        |> error(result, key)
+    end
+  end
```

Comment:
what booleans are we checking here? Rename it so its easy to understand and add single line comment. Also you can use function overloading instead of case statement here, makes it cleaner
like

> defp validate_booleans(result, key, true) do
> defp validate_booleans(result, key, false) do
> defp validate_booleans(result, key, _) do